cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/KafkaAdminTopicConfigProvider.java

```diff
@@ -0,0 +1,171 @@
+        LOG.warn("Cannot retrieve configuration for topic '{}'.", topic, ee);
+      }
+    } catch (InterruptedException ie) {
+      LOG.debug("Interrupted while getting configuration for topic '{}'.", topic, ie);
+    }
+
+    if (topicConfig != null) {
+      return convertTopicConfigToProperties(topicConfig);
+    } else {
+      LOG.warn("The configuration for topic '{}' could not be retrieved, returning empty Properties instance.", topic);
+      return new Properties();
+    }
+  }
+
+  /**
+   * Fetches the configuration for all the topics on the Kafka cluster. If an error is encountered when retrieving the
+   * topic names then the error details will be logged and an empty Map instance will be returned.
+   *
+   * @return A Map from topic name string to Properties instance containing that topic's configuration.
+   */
+  @Override
+  public Map<String, Properties> allTopicConfigs() {
+
+    // Request a map of futures for the config of each topic on the Kafka cluster
+    LOG.debug("Requesting configurations for all topics");
+    Set<String> topicNames = null;
+    try {
+      topicNames = _adminClient.listTopics().names().get();
+    } catch (InterruptedException | ExecutionException e) {
+      LOG.warn("Unable to obtain list of all topic names from the Kafka Cluster");
+    }
+
+    if (topicNames == null) {
+      return new HashMap<>();
+    } else {
+      return topicConfigs(topicNames);
+    }
+  }
+
+  /**
+   * Fetches the configuration for the requested topics. If an error is encountered, for each topic, the details will be
+   * logged and the entry for that topic will be omitted from the returned map.
+   *
+   * @param topics The set of topic names for which the topic-level configurations are required.
+   * @return A Map from topic name string to Properties instance containing that topic's configuration.
+   */
+  @Override
+  public Map<String, Properties> topicConfigs(Set<String> topics) {
+
+    Map<ConfigResource, KafkaFuture<Config>> topicConfigs;
+    topicConfigs = _adminClient.describeConfigs(
+            topics.stream().map(name -> new ConfigResource(ConfigResource.Type.TOPIC, name)).collect(Collectors.toList())
+    ).values();
+
+    Map<String, Properties> propsMap = new HashMap<>();
+    if (topicConfigs != null) {
+      for (Map.Entry<ConfigResource, KafkaFuture<Config>> entry : topicConfigs.entrySet()) {
+        try {
+          Config config = entry.getValue().get();
+          propsMap.put(entry.getKey().name(), convertTopicConfigToProperties(config));
+        } catch (ExecutionException ee) {
+          if (org.apache.kafka.common.errors.TimeoutException.class == ee.getCause().getClass()) {
+            LOG.warn("Failed to retrieve config for topic '{}' due to describeConfigs request time out. Check for Kafka-side issues"
+                            + " and consider increasing the configured timeout.", entry.getKey().name());
+          } else {
```

Comment:
I believe if one of the futures would returne a `org.apache.kafka.common.errors.TimeoutException`, then all of the futures are expected to return a `org.apache.kafka.common.errors.TimeoutException`. Hence, to avoid logging a `warn` for individual topics, maybe we can log a single `warn` message to indicate failure for all topics and then return?